feat: add roundcube template

seppulcro opened the pull request:
Based off https://github.com/Dokploy/dokploy/pull/458 I'd like to also add a template for roundcube so it's easier to get a self-hosted email solution up and running.
Please keep in mind some VPS/ISP block port 25.

SashaGoncharov19 commented on apps/dokploy/templates/roundcube/docker-compose.yml:
```yaml
services:
  roundcubemail:
    image: roundcube/roundcubemail:latest
    volumes:
      - ./www:/var/www/html
      - ./db/sqlite:/var/roundcube/db
    environment:
      - ROUNDCUBEMAIL_DB_TYPE=sqlite
      - ROUNDCUBEMAIL_SKIN=elastic
      - ROUNDCUBEMAIL_DEFAULT_HOST=${DEFAULT_HOST}
      - ROUNDCUBEMAIL_SMTP_SERVER=${SMTP_SERVER}
    networks:
      - dokploy-network
      
networks:
  dokploy-network:
    external: true```

SashaGoncharov19 commented on apps/dokploy/templates/roundcube/index.ts:
```ts
import type { DomainSchema, Schema, Template } from "../utils";
import { generateRandomDomain } from "../utils";

export function generate(schema: Schema): Template {
	const randomDomain = generateRandomDomain(schema);

	const envs = [
		"DEFAULT_HOST=tls://mail.example.com",
		"SMTP_SERVER=tls://mail.example.com",
	];

	const domains: DomainSchema[] = [
		{
			host: randomDomain,
			port: 80,
			serviceName: "roundcubemail",
		},
	];

	return { envs, domains };
}
```

SashaGoncharov19 commented on apps/dokploy/templates/templates.ts:
```ts
{
		id: "roundcube",
		name: "Roundcube",
		version: "1.6.9",
		description:
			"Free and open source webmail software for the masses, written in PHP.",
		logo: "roundcube.svg",
		links: {
			github: "https://github.com/roundcube/roundcubemail",
			website: "https://roundcube.net/",
			docs: "https://roundcube.net/about/",
		},
		tags: ["self-hosted", "email", "webmail"],
		load: () => import("./roundcube/index").then((m) => m.generate),
	},

SashaGoncharov19 reviewed (changes requested):
And run `pnpm run check`

seppulcro commented:
> And run `pnpm run check`

Should be OK in the most recent commit, thanks!

SashaGoncharov19 commented on apps/dokploy/templates/roundcube/docker-compose.yml:
````diff
@@ -0,0 +1,17 @@
+
+networks:
+  dokploy-network:
+    external: true```
````
Wrong characters here

seppulcro replied:
Thanks! Should be OK now

Siumauricio commented:
@seppulcro Can you add also a small explanation in the docs to clarify it needs to have the port 25 open

seppulcro commented:
> @seppulcro Can you add also a small explanation in the docs to clarify it needs to have the port 25 open

Hi! Please check the latest commit :)
I've updated the docs as requested.
While trying to use remark-gfm's footnotes in mdx I noticed the generated id for the url was broken so I fixed that too!

SashaGoncharov19 commented on apps/dokploy/templates/templates.ts:
```diff
@@ -497,4 +497,19 @@ export const templates: TemplateData[] = [
+			website: "https://roundcube.net/",
+			docs: "https://roundcube.net/about/",
+		},
+		tags: ["self-hosted", "mail", "webmail"],
```
we have tag "email", i suggest a change from "mail" -> "email"

seppulcro replied:
Should be OK in the latest commit, do we keep webmail as a "new" tag?
Thanks!

SashaGoncharov19 replied:
Yep, I think what "webmail" can be a new tag. I suggest a change for "email" because we have another templates with this tag.

seppulcro replied:
Yep you're 100% correct. I should've checked on my local what labels were available... 😄

seppulcro commented:
Hey @Siumauricio did you get the chance to look at these changes yet? 😁

Siumauricio commented:
@seppulcro I'm still waiting to you to resolve the comments
![image](https://github.com/user-attachments/assets/71f3ffbf-6b5b-484f-92cc-6991d5261116)

seppulcro commented:
> @seppulcro I'm still waiting to you to resolve the comments ![image](https://private-user-images.githubusercontent.com/47042324/372992593-71f3ffbf-6b5b-484f-92cc-6991d5261116.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc5MzkxNjgsIm5iZiI6MTcyNzkzODg2OCwicGF0aCI6Ii80NzA0MjMyNC8zNzI5OTI1OTMtNzFmM2ZmYmYtNmI1Yi00ODRmLTkyY2MtNjk5MWQ1MjYxMTE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAzVDA3MDEwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3MThjMDk4YjU5ZWYyZWMwOTk3Y2VmMGEwN2Q0NjJhZGQyMzhhOTBjMjQzZjljNWIyYjhmZjRhY2ZmZjUwMjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aOiHzFWZMnvl_Q9BbzghQnNV8OPerrsuqypYIqMFOtE)

That review is pending and I can't see those comments on my end :(
